// @keystone-6/core/___internal-do-not-use-will-break-in-patch/next-graphql/dist/keystone-6-core-___internal-do-not-use-will-break-in-patch-next-graphql.esm.js
import { i as initConfig, c as createSystem } from '../../../dist/initConfig-2c06899a.esm.js';
import { c as createApolloServerMicro } from '../../../dist/createApolloServer-2aead844.esm.js';
import '@babel/runtime/helpers/objectSpread2';
import 'p-limit';
import '../../../dist/core-c6bc4160.esm.js';
import '../../../dist/sqlite-11f2ae03.esm.js';
import '../../../dist/graphql-ts-schema-ecd3b4c2.esm.js';
import '@graphql-ts/schema';
import 'graphql-type-json';
import 'graphql-upload/public/GraphQLUpload.js';
import 'graphql';
import 'decimal.js';
import '@graphql-ts/schema/api-without-context';
import '@graphql-ts/extend';
import '@graphql-ts/schema/api-with-context';
import '@graphql-tools/schema';
import '@babel/runtime/helpers/objectWithoutProperties';
import '../../../dist/graphql-errors-dd21b092.esm.js';
import 'apollo-server-errors';
import 'path';
import 'util';
import 'fs-extra';
import 'resolve';
import '@nodelib/fs.walk';
import '@emotion/hash';
import '../../../dist/admin-meta-graphql-4f7bd0cb.esm.js';
import '@apollo/client';
import '@babel/runtime/helpers/classPrivateFieldInitSpec';
import '@babel/runtime/helpers/classPrivateFieldGet';
import '@babel/runtime/helpers/classPrivateFieldSet';
import '../../../dist/types-for-lists-9f8f5d4c.esm.js';
import 'graphql/execution/values';
import 'pluralize';
import '@babel/runtime/helpers/defineProperty';
import 'uuid';
import 'image-type';
import 'image-size';
import '../../../fields/types/image/utils/dist/keystone-6-core-fields-types-image-utils.esm.js';
import 'crypto';
import 'stream';
import 'filenamify';
import '@sindresorhus/slugify';
import '../../../fields/types/file/utils/dist/keystone-6-core-fields-types-file-utils.esm.js';
import 'node-fetch';
import 'form-data';
import 'cuid';
import '../../../dist/package-path-5ec371a3.esm.js';
import 'apollo-server-micro';
import 'apollo-server-express';
import 'apollo-server-core';
import '../../../session/dist/keystone-6-core-session.esm.js';
import 'cookie';
import '@hapi/iron';
import 'uid-safe';

function nextGraphQLAPIRoute(keystoneConfig, prismaClient) {
  const initializedKeystoneConfig = initConfig(keystoneConfig);
  const {
    graphQLSchema,
    getKeystone
  } = createSystem(initializedKeystoneConfig);
  const keystone = getKeystone(prismaClient);
  keystone.connect();
  const apolloServer = createApolloServerMicro({
    graphQLSchema,
    createContext: keystone.createContext,
    sessionStrategy: initializedKeystoneConfig.session,
    graphqlConfig: initializedKeystoneConfig.graphql,
    connectionPromise: keystone.connect()
  });
  let startPromise = apolloServer.start();
  return async (req, res) => {
    var _keystoneConfig$graph;

    await startPromise;
    return apolloServer.createHandler({
      path: ((_keystoneConfig$graph = keystoneConfig.graphql) === null || _keystoneConfig$graph === void 0 ? void 0 : _keystoneConfig$graph.path) || '/api/graphql'
    })(req, res);
  };
}

export { nextGraphQLAPIRoute };
